pkg/packetfilter/fake/namedset.go

```diff
@@ -0,0 +1,307 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fake
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"github.com/submariner-io/submariner/pkg/packetfilter"
+	"k8s.io/utils/set"
+)
+
+type namedType struct {
+	setInfo *packetfilter.SetInfo
+	pfilter *PacketFilter
+}
+
+func (n *namedType) Name() string {
+	return n.setInfo.Name
+}
+
+func (n *namedType) Flush() error {
+	n.pfilter.flushSet(n.setInfo.Name)
+	return nil
+}
+
+func (n *namedType) Destroy() error {
+	return errors.Wrap(n.pfilter.destroySet(n.setInfo.Name), "Destroy failed")
+}
+
+func (n *namedType) Create(ignoreExistErr bool) error {
+	return errors.Wrap(n.pfilter.createSet(n.setInfo, ignoreExistErr), "Create failed")
+}
+
+func (n *namedType) AddEntry(entry string, ignoreExistErr bool) error {
+	return errors.Wrap(n.pfilter.addEntry(entry, n.setInfo, ignoreExistErr), "AddEntry failed")
+}
+
+func (n *namedType) DelEntry(entry string) error {
+	return errors.Wrap(n.pfilter.delEntry(entry, n.setInfo.Name), "DelEntry failed")
+}
+
+func (n *namedType) ListEntries() ([]string, error) {
+	list, err := n.pfilter.listEntries(n.setInfo.Name)
+	return list, errors.Wrap(err, "ListEntries failed")
+}
+
+func (i *PacketFilter) destroySet(setName string) error {
+	i.mutex.Lock()
+	defer i.mutex.Unlock()
+
+	err := matchForError(&i.failOnDestroySetMatchers, setName)
+	if err != nil {
+		return err
+	}
+
+	if i.sets[setName] == nil {
+		return nil
+	}
+
+	delete(i.sets, setName)
+
+	return nil
```

Comment:
`delete` handles nil

```suggestion
	delete(i.sets, setName)

	return nil
```